src/controller/adc.c

```diff
@@ -42,55 +40,88 @@ void adc_init (void)
   ADC1_ScanModeCmd(ENABLE);
   ADC1_Cmd(ENABLE);
 
-  //********************************************************************************
-  // next code is for "calibrating" the offset value of some ADC channels
 
-  // read and discard few samples of ADC, to make sure the next samples are ok
+
+  // discard few samples of ADC, to make sure the next samples are ok ////////////////////////////////////////
   for(ui8_i = 0; ui8_i < 64; ++ui8_i)
   {
     ui16_counter = TIM3_GetCounter() + 10; // delay ~10ms
-    while(TIM3_GetCounter() < ui16_counter) ; // delay ~10ms
+    
+    // delay ~10ms
+    while(TIM3_GetCounter() < ui16_counter);
+    
+    // trigger ADC conversion on all channels (scan conversion, buffered)
     adc_trigger();
-    while(!ADC1_GetFlagStatus(ADC1_FLAG_EOC)) ; // wait for end of conversion
+    
+    // wait for end of conversion
+    while(!ADC1_GetFlagStatus(ADC1_FLAG_EOC));
   }
 
-  // read and average a few values of ADC battery current
-  ui16_adc_battery_current_offset = 0;
-  for(ui8_i = 0; ui8_i < 16; ++ui8_i)
+
+
+  // ADC battery current initialization //////////////////////////////////////////////////////////////////////
+  
+  uint16_t ui16_adc_battery_current_offset = 0;
+
+  // calibrate battery current
+  for (ui8_i = 0; ui8_i < 16; ++ui8_i)
   {
     ui16_counter = TIM3_GetCounter() + 10; // delay ~10ms
-    while(TIM3_GetCounter() < ui16_counter) ; // delay ~10ms
+    
+    // delay ~10ms
+    while(TIM3_GetCounter() < ui16_counter);
+    
+    // trigger ADC conversion on all channels (scan conversion, buffered)
     adc_trigger();
-    while(!ADC1_GetFlagStatus(ADC1_FLAG_EOC)) ; // wait for end of conversion
+    
+    // wait for end of conversion
+    while(!ADC1_GetFlagStatus(ADC1_FLAG_EOC));
+    
     ui16_adc_battery_current_offset += UI8_ADC_BATTERY_CURRENT;
   }
+  
+  // calculate and set current offset
   ui16_adc_battery_current_offset >>= 4;
   ui8_adc_battery_current_offset = ui16_adc_battery_current_offset >> 2;
   ui8_g_adc_motor_phase_current_offset = ui8_adc_battery_current_offset;
 
-  // read and average a few values of ADC torque sensor
-  ui16_adc_torque_sensor_offset = 0;
-  for(ui8_i = 0; ui8_i < 16; ++ui8_i)
+
+
+  // ADC torque sensor initialization //////////////////////////////////////////////////////////////////////
+  
+  #define ADC_TORQUE_SENSOR_OFFSET_ADJUSTMENT   6
```

Comment:
Would it not be better to move this define to the top of the file ?

Reply:
This is WIP. That #define is supposed to be removed completely in the coming days.

But otherwise it is really a matter of preference. I personally always want everything associated with a function or feature as local as possible and not scattered. But I do agree if it is a #define for many different parts of the code it is better to have it at the top of the file or, even better, in some appropriate header.

Please do let me know if you do not agree or if I have missed something.